js/packages/openinference-vercel/src/utils.ts

```diff
@@ -497,3 +499,41 @@ export const safelyGetOpenInferenceAttributes = withSafety({
   fn: getOpenInferenceAttributes,
   onError: onErrorCallback(""),
 });
+
+const isOpenInferenceSpan = (span: ReadableSpan) => {
+  const maybeOpenInferenceSpanKind =
+    span.attributes[SemanticConventions.OPENINFERENCE_SPAN_KIND];
+  return (
+    typeof maybeOpenInferenceSpanKind === "string" &&
+    maybeOpenInferenceSpanKind in OpenInferenceSpanKind
+  );
+};
+
+/**
+ * Determines whether a span should be exported based on configuration and the spans attributes.
+ * @param span the spn to check for export eligibility.
+ * @param onlyExportOpenInferenceSpans whether or not to only export OpenInference spans.
+ * @returns true if the span should be exported, false otherwise.
+ */
+export const shouldExportSpan = ({
+  onlyExportOpenInferenceSpans,
+  span,
+}: {
+  span: ReadableSpan;
+  onlyExportOpenInferenceSpans: boolean;
+}): boolean => {
+  return !onlyExportOpenInferenceSpans || isOpenInferenceSpan(span);
+};
```

Comment:
Feels like this might be slightly more clean if we had an ExportConfig so this would be:

```typescript
const souldExportSpan = (config: ExportConfig) => (span: ReadableSpan) => boolean
```